src/hawkbit-client.c

```diff
@@ -256,6 +274,24 @@ static void set_default_curl_opts(CURL *curl)
         curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, hawkbit_config->connect_timeout);
         curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, hawkbit_config->ssl_verify ? 1L : 0L);
         curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, hawkbit_config->ssl_verify ? 1L : 0L);
+
+        if (g_once_init_enter(&curl_share_initialized)) {
+                for (gint i = 0; i <= CURL_LOCK_DATA_LAST; i++)
+                        g_mutex_init(&curl_share_locks[i]);
+
+                curl_share = curl_share_init();
+                g_return_if_fail(curl_share);
```

Comment:
We use `g_return_*()` methods for input verification only. Thus better either ignore it (since all potential errors for `curl_share_setopt()` are ignored so far anyway) or use `g_assert_nonnull()`.